report/4.Solution/LanuageOfCompiler.tex

```diff
@@ -0,0 +1,37 @@
+\section{The languages of the compiler}
+When creating a compiler there are three languages of relevance, as see in the tombstone diagram\ref{fig:TStoneExample}.
+
+\begin{figure}[H]
+	\centering
+	\includegraphics[width=\textwidth/2+\textwidth/4]{4.Solution/images/T-diagram.png}
+	\caption{
+		Example of Tombstone diagram showcasing an example of three languages for a given compiler.\cite{TStoneWiki}
+	}
+	\label{fig:TStoneExample}
+\end{figure}
+In the case of the HCL compiler the three languages are as follows:
+
+\begin{align*}
+input \to & HCL \\
+output \to & C++/C \\
+compiler \to & Kotlin \\
+\end{align*}
+
+\subsection{Language of the compiler}
+The compiler was written in Kotlin\cite{KotlinWebsite}, a Java based language.
+This was based on a number of factors.
+The group wanted to experiment with using a language they had no prior experience with for the duration of the project, but because of time limitations, a new language had to adhere to already known paradigms.
```

Comment:
Change first part to: "The group wanted to try a language"
Also (and this is probably useless): "I don't like the they had no prior experience", but don't know what to change it to